front/src/index.html

```diff
@@ -55,133 +61,81 @@
-                        {
-                            "_id": {
-                                "$oid": "607ae876f59d21d5a50ac3c9"
-                            },
-                            "pseudo": "mon pseudo1",
-                            "roomsId": []
-                        },
-                        {
-                            "_id": {
-                                "$oid": "607ae876f59d21d5a50ac3c2"
-                            },
-                            "pseudo": "mon pseudo2",
-                            "roomsId": []
-                        }
-                    ],
-                    messages: [
-                        {_id : {$oid: "459I548383"},content: "Bonjour tata", author: "toto",  timestamp: 1618587750035},
-                        {_id : {$oid: "4459493924"},content: "Salut toto", author: "tata",  timestamp: 1618587750035 },
-                    ]
+                    rooms: [],
+                    users: [],
+                    messages: []
                 }),
+                mounted() {
+                    this.fetchRooms();
+                    setInterval(() => {
+                        this.fetchMessages();
+                    }, 1000);
+                },
                 methods: {
-                    onMessageSend() {
-                        // TODO: implement something that calls the api and send the message
-                        console.debug(`message: ${this.currentTypedMessage}`);
-                        // add the message
-                        this.messages.push({
-                            _id: {
-                                $oid: new Date().getTime().toString(16)
-                            },
-                            author: this.loggedUser.pseudo,
-                            content: this.currentTypedMessage,
-                            timestamp: Math.floor(new Date().getTime())
-                        });
+                    async fetchRooms() {
+                        if(this.loggedUser === null) return;
+                        console.log(this.loggedUser);
+                        this.rooms = await srv.getAllRooms(this.loggedUser)
+                    },
+                    async fetchMessages(){
+                        if(this.selectedRoom != undefined){
+                        this.messages = await srv.getAllMessagesInRoom(this.selectedRoom);
+                        };
+                    },
+                    async onMessageSend() {
+                        await srv.createMessage(this.loggedUser.id, this.currentTypedMessage, this.selectedRoom.id);
                         // reset
                         this.currentTypedMessage = "";
                     },
                     // this is just example code
                     // (called when a room is selected)
-                    displayRoom(roomInfo) {
+                    async displayRoom(roomInfo) {
                         const {room, index} = roomInfo;
```

Comment:
the code in the component should be changed to only provide the room and not the index as it isn't needed